src/middleware/packages/activitypub/services/activitypub/subservices/collection/actions/get.js

```diff
@@ -0,0 +1,275 @@
+
+  do {
+    itemUri = cursorDirection.beforeEq ? allItems.pop() : allItems.shift();
+    if (itemUri) {
+      try {
+        const item = await ctx.call('ldp.resource.get', {
+          resourceUri: itemUri,
+          accept: MIME_TYPES.JSON,
+          jsonContext,
+          webId: webId === 'system' && ctx.meta.impersonatedUser ? ctx.meta.impersonatedUser : webId
+        });
+        selectedItems.push(item);
+      } catch (e) {
+        if (e.code === 404 || e.code === 403) {
+          this.logger.debug(`Item not found with URI: ${itemUri}`);
+        } else {
+          throw e;
+        }
+      }
+    }
+  } while ((selectedItems.length < options.itemsPerPage || !options.itemsPerPage) && itemUri);
+
+  // get the missing cursors for the next/prev links
+  if (cursorDirection.beforeEq) {
+    // also reverse the items in this case (as we pop them in the loop, they are selected in reverse order)
+    selectedItems.reverse();
+    previousItemUri = allItems.pop();
+  } else if (cursorDirection.afterEq) {
+    nextItemUri = allItems.shift();
+  }
+
+  if (!options.dereferenceItems) {
+    selectedItems = selectedItems.map(item => item.id);
+  } else {
+    selectedItems = selectedItems.map(({ '@context': context, ...item }) => item);
+  }
+
+  return {
+    items: selectedItems,
+    previousItemUri,
+    nextItemUri
+  };
+}
+
+function formatResponse(
+  ctx,
+  {
+    items,
+    options,
+    prevCursorUri,
+    nextCursorUri,
+    firstItemUri,
+    lastItemUri,
+    collectionUri,
+    beforeEq,
+    afterEq,
+    localContext
+  }
+) {
+  const itemsProp = options.ordered ? 'orderedItems' : 'items';
+
+  if (options.itemsPerPage && !beforeEq && !afterEq) {
+    // Pagination is enabled but no page is selected, return the collection
+    return {
+      '@context': localContext,
+      id: collectionUri,
+      type: options.ordered ? 'OrderedCollection' : 'Collection',
+      ...options,
+      first: firstItemUri ? `${collectionUri}?afterEq=${encodeURIComponent(firstItemUri)}` : undefined,
+      last: lastItemUri ? `${collectionUri}?beforeEq=${encodeURIComponent(lastItemUri)}` : undefined
+    };
+  }
+
+  if (options.itemsPerPage) {
+    return {
+      '@context': localContext,
+      id: `${collectionUri}?${beforeEq ? 'beforeEq' : 'afterEq'}=${encodeURIComponent(beforeEq || afterEq)}`,
+      type: options.ordered ? 'OrderedCollectionPage' : 'CollectionPage',
+      partOf: collectionUri,
+      prev: prevCursorUri ? `${collectionUri}?beforeEq=${encodeURIComponent(prevCursorUri)}` : undefined,
+      next: nextCursorUri ? `${collectionUri}?afterEq=${encodeURIComponent(nextCursorUri)}` : undefined,
+      [itemsProp]: items
+    };
+  }
+
+  // No pagination, return the collection
+  return {
+    '@context': localContext,
+    id: collectionUri,
+    type: options.ordered ? 'OrderedCollection' : 'Collection',
+    ...options,
+    [itemsProp]: items
+  };
+}
+
+module.exports = {
+  async handler(ctx) {
```

Comment:
Since you use the long format of action (with `handler`), could you also include Moleculer params, like we do here:

https://github.com/assemblee-virtuelle/semapps/blob/master/src/middleware/packages/ldp/services/resource/actions/get.js#L7-L17